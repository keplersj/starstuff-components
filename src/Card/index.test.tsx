import * as React from "react";
import renderer from "react-test-renderer";
import { Card } from ".";

describe("Card", () => {
  describe("basic usage", () => {
    it("renders correctly", () => {
      const tree = renderer.create(<Card title="Test" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with a location", () => {
    it("renders correctly", () => {
      const tree = renderer.create(<Card title="Test" location="#" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with supporting content (in string form)", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(<Card title="Test" supporting="Supporting Content" />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with supporting content (in component form)", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <Card
            title="Test"
            supporting={
              <div>
                <span>Hello</span> <span>world</span>!
              </div>
            }
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with a custom link", () => {
    it("renders correctly", () => {
      const customLink = jest.fn().mockImplementation((title, location) => (
        <span>
          [{title}]({location})
        </span>
      ));

      const tree = renderer
        .create(
          <Card title="Test" location="#" customLinkComponent={customLink} />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
      expect(customLink).toHaveBeenCalledTimes(1);
    });
  });
});
