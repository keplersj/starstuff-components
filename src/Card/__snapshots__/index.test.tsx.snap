// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card basic usage renders correctly 1`] = `
.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<div
  className="emotion-0"
>
  <h2>
    Test
  </h2>
</div>
`;

exports[`Card with a custom link renders correctly 1`] = `
.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<div
  className="emotion-0"
>
  <span>
    [
    Test
    ](
    #
    )
  </span>
</div>
`;

exports[`Card with a location renders correctly 1`] = `
.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<div
  className="emotion-0"
>
  <a
    href="#"
  >
    <h2>
      Test
    </h2>
  </a>
</div>
`;

exports[`Card with supporting content (in component form) renders correctly 1`] = `
.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<div
  className="emotion-0"
>
  <h2>
    Test
  </h2>
  <div>
    <span>
      Hello
    </span>
     
    <span>
      world
    </span>
    !
  </div>
</div>
`;

exports[`Card with supporting content (in string form) renders correctly 1`] = `
.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<div
  className="emotion-0"
>
  <h2>
    Test
  </h2>
  Supporting Content
</div>
`;
